litmus-portal/backend/graphql-server/graph/schema.graphqls

```diff
@@ -144,6 +144,8 @@ input PodLogRequest{
 
 type Query{
   getWorkFlowRuns(project_id: String!): [WorkflowRun!]!
+
+  getCluster(project_id: String!, cluster_type: String!): [Cluster!]!
```

Comment:
Is cluster_type needed? the frontend when querying for all the cluster will not need to make 2 queries for this, we can just fetch all the clusters and show them according to the requirements in the frontend.

Reply:
There is a button called ignite-cluster (internal cluster) to select the target cluster. By clicking on that button an API will be called.

Reply:
To get a list of all the cluster we need to get all the clusters internal as well as external, if we have this query we will have to make 2 separate queries. Only after we list the clusters we can select a cluster for the workflow.

Reply:
This might work for now but later we will have to change this to accommodate external clusters, the `Cluster` response does have a field called cluster_type which can be used in the frontend to determine whether internal or external.

Reply:
In the frontend screen, either he  can select an internal cluster or external cluster

Reply:
![Selection_003](https://user-images.githubusercontent.com/20414564/90221560-07d9e500-de28-11ea-9ef7-90744a74e442.png)

Reply:
internal can be one but there will be many external clusters. If we call all cluster, frontend have to filter it out.

Reply:
I got that part, but when we are listing these clusters we have to query the db, so if we specify the cluster_type we have to make 2 queries, 1 for internal and 1 for external. Which doesn't make much sense because there will always be 1 internal cluster. So we can remove that parameter and only query for all the clusters. Then in the frontend we can detect which one is the internal cluster.

Reply:
we can make the cluster_type not mandatory if cluster_type is not present. it will list all clusters else it will filter based on cluster_type.

Reply:
We will have to show all the clusters here right? Because this page is used by the user when they want to create a workflow for a cluster, that cluster can be any cluster either internal or external.

Reply:
Yes having it not mandatory would be good, that should solve it.

Reply:
not all clusters, if he selects ignite-cluster. he will show the internal cluster.

Reply:
Made the cluster_type not mandatory